### PR TITLE
feat: system os light/dark mode option 🔦

### PR DESCRIPTION
**Pull request by dannyl1u**
## Pull Request Checklist

- [x] **Description:** Briefly describe the changes in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [x] **Documentation:** Have you updated relevant documentation?
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?

---

## Description

Add support for system OS light and dark themes to the web application, improving the user experience and providing a more seamless integration with the user's preferences. Closes https://github.com/open-webui/open-webui/issues/1065 

---

### Added

- Added System option to the theme menu which automatically detects the system's color scheme (light or dark) and applies it as the default theme. This ensures a consistent user experience that aligns with the user's system settings
- Dynamic Theme Switching: The application dynamically updates its theme when the system's color scheme changes. This ensures that the application's appearance remains consistent with the system's settings even when they are altered while the application is running.

### Changed
- Refactor - moved the theme logic to its own `applyTheme` function
- Refactor - add event listener in app.html to update theme automatically if system theme changes
- Refactor - IIFE (Immediately Invoked Function Expression) in app.html to prevent FOUC 
- Refactor - manage theme state using Svelte store

### CODE REVIEW

**Comment by dannyl1u** [comment]
https://github.com/open-webui/open-webui/assets/45186464/ddc57fc7-f6b1-4ce0-8180-1f4757de014b

File: src/lib/components/chat/Settings/General.svelte [hunk 1]
```diff
@@ -18,6 +18,41 @@
 
 	let showAdvanced = false;
 
+	window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => {
```
Comment (tjbck) [inline src/lib/components/chat/Settings/General.svelte:21]: Haven't looked into much yet, but we might want to add this code to `app.html` to prevent FOUC

Comment (dannyl1u) [reply src/lib/components/chat/Settings/General.svelte:21]: Good point I'll take a look

**Comment by dannyl1u** [comment]
> This one any closer to merging @dannyl1u?
found bugs during testing, WIP @justinh-rahb

**Comment by jannikstdl** [comment]
@dannyl1u what are the bugs? Works for me in Win 11.

**Comment by dannyl1u** [comment]
@jannikstdl after setting to theme to **system** in settings, and clicking save, once user returns to settings it will show the theme as either **light** or **dark** when it should be showing **system**

**Comment by dannyl1u** [comment]
Updated to work with the latest UI updates, ready for review
@jannikstdl @justinh-rahb @tjbck 

https://github.com/open-webui/open-webui/assets/45186464/6021549a-e125-447e-a186-399bd1c38431

**Comment by justinh-rahb** [comment]
LGTM!

**Comment by jannikstdl** [comment]
Very nice!

**Comment by Silentoplayz** [comment]
Great job! Thank you very much for implementing this as a feature for all to enjoy! 😀

**Comment by justinh-rahb** [comment]
I'm not sure how many Dark Mode enthusiasts will be suddenly surprised by the darkening of their UI, since I've rarely met one that doesn't immediately go looking for the toggle in any app they use 😆 However this will be one less step for them going forward 🙌

**Comment by tjbck** [comment]
Thanks for the PR! Just did some refac and removed a good majority of unnecessary parts, please test it and let me know if I broke anything.

@dannyl1u Also, I just wanted to mention that it would be great if we could stick to the conventions already used in our repo. I realise I should have provided more information about these conventions earlier, but couldn't find the time 😅. You can use the refactored code as a reference for future work. Please feel free to review the changes I've made, and if you have any questions, don't hesitate to reach out!

**Comment by dannyl1u** [comment]
LGTM!

**Pull request merged**